math/average.c

```diff
@@ -0,0 +1,25 @@
+#include <stdio.h>
+#include <assert.h>
+
+//average of an array
+
+double average(int avg[]) {
+    double result = 0;
+    int size=0;
+    
+    while(avg[size+1]!='\0'){
+        size++;
+    }
+
+    for (int i = 0; i < size; i++) {
+        result += avg[i];
+    }
+
+    return result / size;
+}
+
+int main() {
+    int test[] = {1,2,3,4,5};
```

Comment:
Another comment in regards to testing, there is a common testing documentation format called [gherkin](https://cucumber.io/docs/gherkin/reference/) The basics of it are 

```
// Given we have an array fo values

// When we try to work out the average value of the array

// Then we should get the expected result
```

This is a simple example but simple is good.